[JerryWang0000] iP

JerryWang0000 opened the pull request:
# Duke

> "Yes we're all using it." – Eric Cartman ([source](https://www.youtube.com/watch?v=6lMaepNvP6A&t=89s))

Duke is a simple tool for task management and saves your effort of remembering that much stuff. It can: 

- tell you what need to do
- check if something is already done
- delete stuff you don't care any more

You just need to
1. Extract the Jar file
2. Run it just like how you run every Jar file :rocket:

You can use it for **FREE**.

Features:

- [X] Manage your tasks
- [X] Search by dates
- [ ] Modifying the content of tasks(Coming soon)

If you are a Java Programmer, here is the 'main()' method for your reference:

```Java
public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

JerryWang0000 committed:
Change name of the bot and add greeting and exit words

Rewrite the src/main/java/Duke.java to allow for the changes mentioned in the subject line.

JerryWang0000 committed:
Add BotManager.java and modify Duke.java

Create a BotManager class to take care of all reponses that will be generated by the bot. Some codes in the previous commit is moved to the BotManager class.

JerryWang0000 committed:
Add "add" and "list" function to the bot

Modify the void answer(String prompt) function to allow the bot to add tasks and list all the tasks added.

JerryWang0000 committed:
Add task.java and modify BotManager.java

Create a new Task class to facilitate the increasingly complicated features, and make amendment to the BotManager in accordance.

JerryWang0000 committed:
Add 3 subclasses of Task

Add the subclasses as they present similar behavior. Rewrite some methods in BotManager for better clarity. Make Task class abstract as only the subclasses are dealt with.

JerryWang0000 committed:
Modify input.txt and EXPECTED.TXT

Modify the test files to facilitate the varying features of the bot. Make mild changes to the BotManager.java to adapt to the comparisons.

JerryWang0000 committed:
Add DukeException.java. Make abundant modification to BotManager.java

Add DukeException class that deals with invalid inputs. Change much in the implementation of BotManager to catch DukeException. Add an extra interface to Task class to consider marginal scenarios.

This version needs further improvement to be more punctual when dealing with task creation. Should make distinction between information missing and incorrect format.

JerryWang0000 committed:
Change BotManager.java

Add a deleteTask function to allow for deleting tasks from the collection.

itstrueitstrueitsrealitsreal commented on src/main/java/BotManager.java:
```diff
@@ -0,0 +1,150 @@
+
+public class BotManager {
+    private final String name;
+    private ArrayList<Task> tasks;
```
I would consider making tasks a final attribute since you shouldn't be reassigning it later on.
```suggestion
    private final ArrayList<Task> tasks;
```

itstrueitstrueitsrealitsreal commented on src/main/java/BotManager.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    void greeting() {
+        System.out.println("    Hello! I'm " + name);
```
I like how you abstracted the name out so you can change it later on!

itstrueitstrueitsrealitsreal commented on src/main/java/BotManager.java:
```diff
@@ -0,0 +1,150 @@
+        this.tasks = new ArrayList<Task>();
+    }
+
+    void greeting() {
```
I would change the name of the function to be greet instead of greeting in line with the coding standard.
```suggestion
    void greet() {
```

headcube1 replied:
Nice try on using verbs for methods, however maybe changing to greet would have been better?

itstrueitstrueitsrealitsreal commented on src/main/java/BotManager.java:
```diff
@@ -0,0 +1,150 @@
+        String[] order = prompt.split(" ", 2);
+        String taskType = order[0];
+        switch (taskType) {
+        case "todo":
```
Like that you adhered to the coding standard for something like switch case statements, a lot of my peers overlooked this.

itstrueitstrueitsrealitsreal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,19 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        BotManager manager = new BotManager("Jerry");
```
Interesting decision to name the bot after yourself...

itstrueitstrueitsrealitsreal commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+abstract public class Task {
+    private final String name;
+    private boolean done;
```
I would suggest that done be changed to isDone to make it sound more like a boolean, according to the coding standard for this course.
```suggestion
    private boolean isDone;
```

headcube1 replied:
Changing to isDone would have been better to follow the standard? For other boolean variables as well!

itstrueitstrueitsrealitsreal reviewed (commented):
Other than a few nits, LGTM!

headcube1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+        this.done = false;
+    }
+
+    boolean isDone() {
```
Nicely done in naming the boolean method as isDone()!

headcube1 commented on src/main/java/BotManager.java:
```diff
@@ -0,0 +1,150 @@
+import java.util.ArrayList;
+
+public class BotManager {
```
In general the names of the methods and variables follow the coding standard!

headcube1 reviewed (commented):
Nice work!

JerryWang0000 committed:
Add File attribute to the BotManager

Make mass modification to the code in BotManager.java to adapt to the new attributes. Any change of the current tasks invokes change in the file. Use the file to store tasks given a certain format.

Ignore the temporarily generated duke.txt file.

JerryWang0000 committed:
Merge branch 'branch-Level-7'

JerryWang0000 committed:
Modify Task.java and its subclasses

Modify the type of date-related strings to a specialized LocalDateTime class to manage the timestamps of tasks. Modify the toString() and file I/O methods to adapt to these changes. Add searchDate(LocalDate) function to BotManager to alllow users to search tasks by specific dates.

Make mild changes to satisfy the coding standards.

JerryWang0000 committed:
Merge branch 'branch-Level-8'

JerryWang0000 committed:
Refactor the codes to make the project more OOP-styled

Add Parser class to analyze user's input. Add Storage class to deal with file management. Add TaskList class to temporary store all tasks during the execution of program. Add Ui class for user interaction.

Make changes to other parts of codes to adapt to these major changes. Also make minor changes in accordance with the coding standard.

JerryWang0000 committed:
Merge branch 'branch-A-MoreOOP'

JerryWang0000 committed:
Organize all java files into 2 packages.

The command classes and other class files are kept separately.

JerryWang0000 committed:
Merge branch 'branch-A-Packages'

JerryWang0000 committed:
Merge branch 'add-gradle-support' of https://github.com/JerryWang0000/ip

JerryWang0000 committed:
Add JUnit test codes

Test Parser.parse to see whether it can deliver commands correctly. Test Event.matchDate to see if it can filter events with the corresponding dates.

Override equals method for command classes to make comparisons.

JerryWang0000 committed:
Merge branch 'branch-A-JUnit'

JerryWang0000 committed:
Modify build.gradle to support JAR file creation

JerryWang0000 committed:
Merge branch 'branch-A-Jar'

JerryWang0000 committed:
Add JavaDoc files to all the non-private methods and classes.

JerryWang0000 committed:
Make minor to the codes to make them satisfy the coding standards

JerryWang0000 committed:
Add SearchDateCommand.java

Add SearchDateCommand to allow for searching tasks with certain keywords. Make minor changes to other parts of codes to adapt to the new feature.

JerryWang0000 committed:
Merge branch 'branch-A-JavaDoc'

JerryWang0000 committed:
Merge branch 'branch-A-Level-9'

JerryWang0000 committed:
Merge branch 'branch-A-CodingStandard'

JerryWang0000 committed:
Add GUI to the whole program

Add GUI to the program using FXML. Make mass change to the code. Reformat the logics.

JerryWang0000 committed:
Merge branch 'branch-Level-10'

JerryWang0000 committed:
Add gui package

Add a new package to deal with GUI implementation. Modify getResponse(String input) in duke.Duke to allow for more conversation. Make modification to other parts of the code accordingly.

JerryWang0000 committed:
Merge branch 'branch-Level-10'

JerryWang0000 committed:
Add Assert feature

Add "assert" keyword in the code to increase readability. Make Minor changes in the coding format to satisfy the coding standards.

JerryWang0000 committed:
Increase the quality of code

Modify the header comments in several parts of the codes to increase readability.

JerryWang0000 committed:
Merge pull request #3 from JerryWang0000/branch-A-Assertions

Add Assert feature.

JerryWang0000 committed:
Merge branch 'master' into branch-A-CodeQuality

JerryWang0000 committed:
Merge pull request #2 from JerryWang0000/branch-A-CodeQuality

Increase the quality of codes

JerryWang0000 committed:
Merge branch 'master' of https://github.com/JerryWang0000/ip

* 'master' of https://github.com/JerryWang0000/ip:
  Increase the quality of code

JerryWang0000 committed:
Modify the search date feature

Make modification to the original searchDate function in TaskList to sort all tasks in chronological order, and include all Todo tasks.

Add a new TsakWithTime interface to allow for comparisons among tasks and conform to ISP.

Modify Parse.parse to reduce ambiguity.

JerryWang0000 committed:
Merge pull request #4 from JerryWang0000/branch-B-ViewSchedules

Modify the search date feature

JerryWang0000 committed:
Merge branch 'master' of https://github.com/JerryWang0000/ip

* 'master' of https://github.com/JerryWang0000/ip:
  Modify the search date feature

JerryWang0000 committed:
Enhance the GUI

Make modification to the FXML file to renovate the layout of GUI. Add a title to the stage to make the program more personalized.

JerryWang0000 committed:
Merge branch 'branch-A-BetterGui'

JerryWang0000 committed:
Make minor changes to the layout of DialogBox

JerryWang0000 committed:
Update README.md

Update README.md to include a user guide to the program. Make minor changes to other parts of the codes to allow for more flexibility.

JerryWang0000 committed:
Update README.md

JerryWang0000 committed:
Update DialogBox.fxml

JerryWang0000 committed:
Merge branch 'master' of https://github.com/JerryWang0000/ip

JerryWang0000 committed:
Update README.md

JerryWang0000 committed:
Fix a major bug in the previous release

Modify the output string in multiple commands to show full message in the GUI. Mofidy the Ui.png to reflect these changes.

JerryWang0000 committed:
Update Ui.png
